client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -640,3 +642,88 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def get_export_presets_by_filtering(export_preset_name, channel_type_list):
```

Comment:
I wonder if this function by itself does too much at once?

It:
- Gets a resource preset by 'nice name'
- Creates a export preset dict for that preset with the specified channel type names

Should this instead take a Subtsance Painter Resource Preset instance directly, e.g.

```python
def get_export_preset_by_name(preset_name: str) -> substance_painter.export.Resource:
    export_presets= get_export_presets()
    preset_full_name = export_presets[preset_name]
    for export_preset in substance_painter.export.list_resource_export_presets():
        if export_preset.resource_id.name == preset_full_name:
            return export_preset


def get_export_preset_with_filtered_maps(
    export_preset: substance_painter.export.Resource, 
    channel_type_names: list[str]
) -> dict:
    filtered_maps = []
    for output_map in export_preset.list_output_maps():
        output_filename = output_map.get("fileName")
        if not output_filename:
            continue
        
        if any(
            channel_type_name in output_filename
            for channel_type_name in channel_type_names
        ):
            filtered_maps .append(channel_map)
    return {
        ...  # return the dict format you need
    }
```